src/Owin.Security.Providers.WSO2/WSO2AuthenticationExtensions.cs

```diff
@@ -0,0 +1,31 @@
+﻿using Microsoft.Owin;
+using Owin;
+using System;
+
+namespace Owin.Security.Providers.WSO2
+{
+    public static class WSO2AuthenticationExtensions
+	{
+		public static IAppBuilder UseWSO2Authentication(this IAppBuilder app, WSO2AuthenticationOptions options)
+		{
+			if (app == null)
+				throw new ArgumentNullException(nameof(app));
+			if (options == null)
+				throw new ArgumentNullException(nameof(options));
+
+			app.Use(typeof(WSO2AuthenticationMiddleware), app, options);
+
+			return app;
+		}
+
+		public static IAppBuilder UseWSO2Authentication(this IAppBuilder app, string baseUrl, string clientId, string clientSecret)
+		{
+			return app.UseWSO2Authentication(new WSO2AuthenticationOptions
+			{
+				BaseUrl = baseUrl,
```

Comment:
I agree that we need a better comments for the parameters, this should make the use of the api much easier. We also should make sure the uri ending with "/". I will update soon.
However I do think it's not quite necessary to use uri type instead of string as uri doesn't guarantee the value is a well formed url as well. Also we don't need to enforce https as the base url is actually the url of WSO2 identity server, which is configurable. The user should have known it already. If they put a wrong url accidentally the error message will tell the details. 

What do you think?